tests/software_tests/database/data_record/test_mapping_data_record.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+from mock import Mock, patch
+
+from uds.database.data_record.mapping_data_record import MappingDataRecord
+
+SCRIPT_LOCATION = "uds.database.data_record.mapping_data_record"
+
+class TestTextTableDataRecord:
+
+    def setup_method(self):
+        self.mock_data_record = Mock(spec=MappingDataRecord)
+
+    @pytest.mark.parametrize(
+        "name, length, mapping", [
+            ("TestMappingDataRecord", 32, {1: "foo", 2: "bar"}),
+            (Mock(), Mock(), Mock()),
+        ]
+    )
+    @patch(f"{SCRIPT_LOCATION}.RawDataRecord.__init__")
+    def test_init__valid(self, mock_raw_data, name, length, mapping):
+        assert MappingDataRecord.__init__(self.mock_data_record, name, length, mapping) is None
+        mock_raw_data.assert_called_once_with(name, length)
+        assert self.mock_data_record.length == length
```

Comment:
please add one more assert (checking that mapping is properly set):
`assert self.mock_data_record.mapping == mapping`